libcudacxx/include/cuda/std/__cccl/extended_data_types.h

```diff
@@ -81,14 +82,15 @@
 #  define _CCCL_HAS_NVFP4() 1
 #endif
 
-// NVC++ supports float128 only in host code
-#if !defined(CCCL_DISABLE_FLOAT128_SUPPORT) && _CCCL_OS(LINUX)                     \
-  && ((_CCCL_COMPILER(NVRTC) && defined(__CUDACC_RTC_FLOAT128__)) /*NVRTC*/        \
-      || defined(__SIZEOF_FLOAT128__) || defined(__FLOAT128__)) /*HOST COMPILERS*/ \
-  && (!defined(__CUDA_ARCH__) || (defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 1000)) /*DEVICE CODE*/
-#  undef _CCCL_HAS_FLOAT128
-#  define _CCCL_HAS_FLOAT128() 1
-#endif
+#if !defined(CCCL_DISABLE_FLOAT128_SUPPORT) && _CCCL_OS(LINUX) && !_CCCL_ARCH(ARM64)
+#  if (defined(__CUDACC_RTC_FLOAT128__) || defined(__SIZEOF_FLOAT128__) || defined(__FLOAT128__)) /*HOST COMPILERS*/
+#    if _CCCL_CUDA_COMPILER(NVHPC) \
+      || ((_CCCL_CUDA_COMPILER(NVCC) || _CCCL_CUDA_COMPILER(CLANG)) && __CUDA_ARCH__ >= 1000) /*DEVICE CODE*/
```

Comment:
@dkolsen-pgi any chance that we can check device architecture here for NVHPC?

Reply:
NVC++ predefines the macro `NV_TARGET_SM_INTEGER_LIST`, which is a space-separated list of all the two-digit architectures in this compilation, similar to the macro `__CUDA_ARCH_LIST__` that NVCC defines.  You can use either of the macros `NV_TARGET_MINIMUM_SM_INTEGER` or `__CUDA_MINIMUM_ARCH__` from `<nv/target>` to get the minimum value from that list as a two-digit value or a three-digit value respectively.

(Remember that NVC++ is a single-pass compiler, so there aren't separate preprocessor passes for the different GPU architectures.)

Reply:
But in this case, how to figure out the GPU architecture when compiler with NVC++ doesn't matter, since NVC++ doesn't support 128-bit floating-point in device code at all, regardless of the GPU architecture.